app/controllers/home_controller.rb

```diff
@@ -0,0 +1,13 @@
+require 'product_filter_service'
+
+class HomeController < ApplicationController
+  def index
+    @products = Product.all
+  end 
+  
+  def apply_filters
+    product_filter_service = ProductFilterService.new(params)
+    @products = product_filter_service.filter_products
+    render :index
+  end
```

Comment:
I would combine this with `index` and apply filters only if there are params for filtering